tap-agent/src/agent/sender_accounts_manager.rs

```diff
@@ -129,18 +122,18 @@ impl Actor for SenderAccountsManager {
                 'scalar_tap_receipt_notification'",
             );
         let myself_clone = myself.clone();
-        let mut accounts_clone = escrow_accounts.clone();
-        let _eligible_allocations_senders_handle = tokio::spawn(async move {
-            while accounts_clone.changed().await.is_ok() {
-                myself_clone
-                    .cast(SenderAccountsManagerMessage::UpdateSenderAccounts(
-                        accounts_clone.borrow().get_senders(),
-                    ))
+        let accounts_clone = escrow_accounts.clone();
+        let _eligible_allocations_senders_handle =
+            watch_pipe(accounts_clone, move |escrow_accounts| {
+                let myself = myself_clone.clone();
```

Comment:
Since you are not sending it to the async block, you don't need to clone it here again.